buttercup/cogs/history.py

```diff
@@ -92,6 +92,18 @@ def add_rank_lines(ax: plt.Axes, gamma: int) -> plt.Axes:
     return ax
 
 
+def add_rate_lines(ax: plt.Axes, max_rate: int) -> plt.Axes:

```

Comment:
This code is quite similar to 85-92, I feel like there should be some generalization possible there.

Reply:
True, I generalized it a bit.